File: frontend/src/types/comment.ts
import type { Audience } from "@/types/insight";

export interface Comment {
  id: string;
  company_id: string;
  period: string;
  audience: Audience;
  user_id: string | null;
  author_name: string;
  author_avatar_url: string | null;
  content: string;
  edited: boolean;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/comment.py
import uuid
from datetime import date, datetime

from pydantic import Field

from app.models.enums import Audience
from app.schemas.base import AppBaseModel


class CommentCreate(AppBaseModel):
    period: date
    audience: Audience
    content: str = Field(..., min_length=1)


class CommentUpdate(AppBaseModel):
    content: str = Field(..., min_length=1)


class CommentRead(AppBaseModel):
    id: uuid.UUID
    company_id: uuid.UUID
    period: date
    audience: str
    user_id: uuid.UUID | None
    author_name: str
    author_avatar_url: str | None = None
    content: str
    edited: bool
    created_at: datetime
    updated_at: datetime
